presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java

```diff
@@ -1,8 +1,284 @@
+                //{storageFormat("PARQUET", ImmutableMap.of("hive.parquet_optimized_reader_enabled", "true"))},
+                //{storageFormat("RCBINARY")},
+                //{storageFormat("RCTEXT")},
+                //{storageFormat("SEQUENCEFILE")},
+                //{storageFormat("TEXTFILE")}
+        };
+    }
+
+    @Test(dataProvider = "storage_formats", groups = {STORAGE_FORMATS})
+    public void testInsertIntoTable(StorageFormat storageFormat)
+    {
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_test_insert_into_" + storageFormat.getName().toLowerCase();
+
+        // DROP TABLE
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+
+        // CREATE TABLE
+        String createTable = format(
+                "CREATE TABLE %s(" +
+                        "   orderkey      BIGINT," +
+                        "   partkey       BIGINT," +
+                        "   suppkey       BIGINT," +
+                        "   linenumber    BIGINT," +
+                        "   quantity      BIGINT," +
+                        "   extendedprice DOUBLE," +
+                        "   discount      DOUBLE," +
+                        "   tax           DOUBLE," +
+                        "   linestatus    VARCHAR," +
+                        "   shipinstruct  VARCHAR," +
+                        "   shipmode      VARCHAR," +
+                        "   comment       VARCHAR," +
+                        "   returnflag    VARCHAR" +
+                        ") WITH (format='%s')",
+                tableName,
+                storageFormat.getName());
+        query(createTable);
+
+        // INSERT INTO TABLE
+        String insertInto = format("INSERT INTO %s " +
+                "SELECT " +
+                "orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, " +
+                "linestatus, shipinstruct, shipmode, comment, returnflag " +
+                "FROM tpch.%s.lineitem", tableName, TPCH_SCHEMA);
+        query(insertInto);
+
+        // SELECT FROM TABLE
+        assertSelect("select sum(tax), sum(discount), sum(linenumber) from %s", tableName);
+
+        // DROP TABLE
+        query(format("DROP TABLE %s", tableName));
+    }
+
+    @Test(dataProvider = "storage_formats", groups = {STORAGE_FORMATS})
+    public void testCreateTableAs(StorageFormat storageFormat)
+    {
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_test_create_table_as_select_" + storageFormat.getName().toLowerCase();
+
+        // DROP TABLE
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+
+        // CREATE TABLE AS SELECT
+        String createTableAsSelect = format(
+                "CREATE TABLE %s WITH (format='%s') AS " +
+                        "SELECT " +
+                        "orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, " +
```

Comment:
`dwrf` format does not support the `DATE` columns. This test is not supposed to check all the column types, but rather the READ/WRITE mechanisms in general for all the storage formats from end to end. All supported column types are already checked in `TestHiveStorageFormats` integration test.